yellowbrick/features/importances.py

```diff
@@ -0,0 +1,250 @@
+        if self.relative:
+            maxv = self.feature_importances_.max()
+            self.feature_importances_ /= maxv
+            self.feature_importances_ *= 100.0
+
+        # Create labels for the feature importances
+        # NOTE: this code is duplicated from MultiFeatureVisualizer
+        if self.labels is None:
+            # Use column names if a dataframe
+            if is_dataframe(X):
+                self.features_ = np.array(X.columns)
+
+            # Otherwise use the column index as the labels
+            else:
+                _, ncols = X.shape
+                self.features_ = np.arange(0, ncols)
+        else:
+            self.features_ = np.array(self.labels)
+
+        # Sort the features and their importances
+        sort_idx = np.argsort(self.feature_importances_)
+        self.features_ = self.features_[sort_idx]
+        self.feature_importances_ = self.feature_importances_[sort_idx]
+
+        # Draw the feature importances
+        self.draw()
+        return self
+
+    def draw(self, **kwargs):
+        """
+        Draws the feature importances as a bar chart; called from fit.
+        """
+        # Quick validation
+        for param in ('feature_importances_', 'features_'):
+            if not hasattr(self, param):
+                raise NotFitted("missing required param '{}'".format(param))
+
+        # Find the positions for each bar
+        pos = np.arange(self.features_.shape[0]) + 0.5
+
+        # Plot the bar chart
+        self.ax.barh(pos, self.feature_importances_, align='center')
+
+        # Set the labels for the bars
+        self.ax.set_yticks(pos)
+        self.ax.set_yticklabels(self.features_)
+
+        return self.ax
+
+    def finalize(self, **kwargs):
+        """
+        Finalize the drawing setting labels and title.
+        """
+        # Set the title
+        self.set_title('Feature Importances of {} Features using {}'.format(
+                len(self.features_), self.name))
+
+        # Set the xlabel
+        xlabel = 'relative' if self.relative else 'feature'
+        self.ax.set_xlabel("{} importance".format(xlabel))
+
+        # Remove the ygrid
+        self.ax.grid(False, axis='y')
+
+        # Ensure we have a tight fit
+        plt.tight_layout()
+
+    def _find_importances_param(self):
+        """
+        Searches the wrapped model for the feature importances parameter.
+        """
+        for attr in ("feature_importances_", "coef_"):
+            try:
+                return getattr(self.estimator, attr)
+            except AttributeError:
+                continue
+
+        raise YellowbrickTypeError(
+            "could not find feature importances param on {}".format(
+                self.estimator.__class__.__name__
+            )
+        )
+
+    def _is_fitted(self):
+        """
+        Returns true if the visualizer has been fit.
+        """
+        return hasattr(self, 'feature_importances_') and hasattr(self, 'features_')
```

Comment:
maybe this should be moved into utils.helpers.py

Reply:
Yeh, this is something I do at work with the estimators I make. Basically, this is an estimator-specific definition of what being fitted means. I do have utility code that goes something like this:

```python
import sklearn 
import warnings 
from sklearn.utils.validation import check_is_fitted

def is_fitted(estimator):
    if hasattr(estimator, '_is_fitted'):
         return estimator._is_fitted() 

    warnings.warn("performing a broad search of fit params", UserWarning) 
    try:
         check_is_fitted(estimator, ["coef_", "estimator_"], all_or_any="any")
         return True 
    except sklearn.exceptions.NotFittedError:
          return False 
```

The problem is that you have to guess what the internal params are for `check_is_fitted` to work. I'm all for bringing this into Yellowbrick though, I'll do another PR that basically wraps the sklearn method and raises a yellowbrick exception.